### PR TITLE
Shadow the jetcd-all

### PR DESCRIPTION
**Pull request by vinsia**
### Background
This PR is going to provide a shadow `jetcd-core` package. The following dependencies will be shaded:
1. `io.grcp.*`
2. `io.netty.*`

The above two packages are often imported by other projects. Besides, shadow all the related SPI files. like `META-INF/io.grpc.NameResolverProvider`

### Testing
Have manually tested with this package, it works well. with conflict packages that mentioned above.

### CODE REVIEW

**Comment by vinsia** [comment]
@lburgazzoli PTAL, thanks

**Comment by vinsia** [comment]
@lburgazzoli ping

**Review by lburgazzoli — CHANGES_REQUESTED** [review]
I would like to have a dedicated module instead of having a shaded jar part of jetcd-core

**Comment by yijianli** [comment]
use a dedicated jetcd repo

**Comment by yijianli** [comment]
@lburgazzoli PTAL

File: build.gradle [hunk 1]
```diff
@@ -81,5 +81,7 @@ subprojects {
         theme 'mocha-parallel'
         showStandardStreams false
     }
+
+    tasks.withType(Javadoc).all { enabled = false }
```
Comment (lburgazzoli) [inline build.gradle:85]: what is this about ?

File: jetcd-all/build.gradle [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+dependencies {
+    api project(':jetcd-core')
+}
+
+shadowJar {
+    relocate 'io.grpc', 'io.etcd.jetcd.shaded.io.grpc'
+    relocate 'io.netty', 'io.etcd.jetcd.shaded.io.netty'
+    mergeServiceFiles()
+
+    archiveClassifier.set('')
```
Comment (lburgazzoli) [inline jetcd-all/build.gradle:31]: what is this about ?

Comment (vinsia) [reply jetcd-all/build.gradle:31]: by default, the shadow jar will add the classifier `all` to the end. so the package name will be `jetcd-all-0.7.1-SNAPSHOT-all.jar`. This command will try to remove the classifier. So the name will be `jetcd-all-0.7.1-SNAPSHOT.jar` eventually.

File: build.gradle (hunk 1 above)
Comment (vinsia) [reply build.gradle]: removed, the java doc is not working in my side previously.
Test again, it should works now.

File: jetcd-all/build.gradle (hunk 2 above)
Comment (vinsia) [reply jetcd-all/build.gradle:31]: I thought since we are using a dedicated repo, we do not need the classifier now.

**Pull request closed without merging**